### PR TITLE
fix: datagrid custom filter for 2nd column doesn't work

### PR DESCRIPTION
**Pull request by kathygit**
## PR Checklist

Please check if your PR fulfills the following requirements:

* [n] Tests for the changes have been added (for bug fixes / features)
* [n/a] Docs have been added / updated (for bug fixes / features)
* [n/a] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?
Before the fix, custom filter doesn't  work for the 2nd column. 

Issue Number: 3486
https://github.com/vmware/clarity/issues/3486

## What is the new behavior?
After the fix, custom filter will work for all columns including the 2nd column. 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

Signed-off-by: Kathy Laird <kathy.laird@emc.com>

### CODE REVIEW

**Comment by kathygit** [comment]
tests passed -- 

  Datagrid
    Providers
      FiltersProvider provider
        ✔ detects if it has active filters
        ✔ can return a list of active filters
        ✔ ignores inactive filters
        ✔ uses all active filters
        ✔ exposes an Observable that proxies all filters changes
        ✔ un-registers an inactive filter
        ✔ un-registers an active filter
        ✔ correctly updates hasUnregistered property

Finished in 3.257 secs / 0.043 secs @ 00:02:28 GMT-0600 (Mountain Daylight Time)

**Comment by bdryanovski** [comment]
@kathygit thanks for the PR.

Could do some changes:
 * Provide unit test for this bugfix
 * Sign off your commit
 
Thanks again :)

**Comment by kathygit** [comment]
@bdryanovski 
Thx! Signoff is added to the last line of the commit comment ? Is that done correct?
I will add tests.

**Comment by coryrylan** [comment]
@kathygit here is an example of the commit format to get the signoff to pass:

```
fix: datagrid custom filter for 2nd column doesn't work

closes #3486

Signed-off-by: Kathy Laird <kathy.laird@emc.com>
```

**Comment by kathygit** [comment]
Test is added. Pls review

**Comment by kathygit** [comment]
I don't know how to change the comments for an existing commit ?

**Comment by hippee-lee** [comment]
@kathygit you can amend your commit to sign it and update the message like this:

```git commit --amend --author "Kathy Laird <kathy.laird@emc.com>"```

Then, in the new commit message make sure you add the line `Signed-off-by: Kathy Laird kathy.laird@emc.com` 

You will have to force push it to your branch.

**Comment by kathygit** [comment]
Thank you, it fixes the last commit, how to amend commit message for the 1st commit ?

**Comment by kathygit** [comment]
@bdryanovski pls review 
I added tests.
And, signed both commits.

File: src/clr-angular/data/datagrid/providers/filters.ts [hunk 1]
```diff
@@ -68,7 +68,10 @@ export class FiltersProvider<T = any> {
         return;
       }
       subscription.unsubscribe();
-      this._all.splice(index, 1);
```
[the remark this answers is not in the document]
Comment (kathygit) [reply src/clr-angular/data/datagrid/providers/filters.ts]: will do

**Comment by hippee-lee** [comment]
@kathygit - you only need to worry about the last commit - all previous commits will get squashed into one before the PR is merged. 

**git cli tip**: `git commit -S -s -m` will sign a commit with the configured username and email address for that repo (or globally depending on how your system is set up)

```
git config --global user.name   # global system user name
git config --global user.email  # global user email
git config user.name            # local user name for a specific repo
git config user.email           # local user email for a specific email
```

File: src/clr-angular/data/datagrid/providers/filters.spec.ts [hunk 2]
```diff
@@ -127,6 +127,32 @@ export default function(): void {
       expect(filters.length).toBe(2);
     });
   });
+
+  describe('FiltersProvider provider unregisters filters correctly', function() {
```
[the remark this answers is not in the document]
Comment (kathygit) [reply src/clr-angular/data/datagrid/providers/filters.spec.ts]: It is a cleaner test where it focused on unregister and check what is left after the unregister

**Comment by bdryanovski** [comment]
You will need to rebase the commits. There is info on the documentation that you could find [here](https://github.com/vmware/clarity/blob/master/DEVELOPMENT_CONTRIBUTION.md#shipping-it)

But the short version is that you will need to squash all the commits into one. And be sure to add the Signed-off into the description of the commit.

```bash
git rebase -i master
```
Into the interactive version, you will have to mark the last two commits to be squashed into the first one. And update the message to match the commit guideline that you could find [here](https://github.com/vmware/clarity/blob/master/DEVELOPMENT_CONTRIBUTION.md#commits)

The last step will be to force push the changes to your fork. 

That's all for the commit part.

**Comment by kathygit** [comment]
did rebase and push to master ... 
What is next ? 
Thx!

**Comment by kathygit** [comment]
@gnomeontherun I did squash just now. Please review if it is all good now. 
Sorry, I have not that good fancy git command :-)

**Comment by gnomeontherun** [comment]
Perfect thanks! Once tests pass and everything is green we can merge on our end.

**Comment by amellnik** [comment]
Looks great @kathygit!  

Clarity team: would it be possible to cut a new release after this is merged to get this and the numeric filter fix out?

**Comment by kathygit** [comment]
@gnomeontherun I wonder when the next release will be? what version shall I watch for it ?

**Comment by gnomeontherun** [comment]
Once we get it passing, I expect it will go out in this weeks release.

**Comment by namitabandal** [comment]
I am facing the same issue in clarity version 3.1.5
In which version should I get the fix on this issue?

**Pull request merged**